draft-pda-protocol.md

```diff
@@ -1057,11 +1078,40 @@ order to achieve these goals:
 
 1. Minimum batch size. The aggregation batch size has an obvious impact on
    privacy. (A batch size of one hides nothing of the input.)
-   {{questions-and-params}} discusses appropriate batch sizes and how they
-   pertains to privacy in more detail.
 2. Aggregation function choice. Some aggregation functions leak slightly more
-   than the function output itself. {{questions-and-params}} discusses the
-   leakage profiles of various aggregation functions in more detail.
+   than the function output itself.
+
+[TODO: discuss these in more detail.]
+
+## Infrastructure diversity
+
+Prio deployments should ensure that aggregators do not have common dependencies
+that would enable a single vendor to reassemble inputs. For example, if all
+participating aggregators stored unencrypted input shares on the same cloud
+object storage service, then that cloud vendor would be able to reassemble all
+the input shares and defeat privacy.
+
+## Data integrity constraints
+
+Data integrity concerns the accuracy and correctness of the outputs in the
+system. The integrity of the output can be influenced by an incomplete round of
+aggregation caused by network partitions, or by bad actors attempting to cause
+inaccuracies in the aggregates. An example data integrity constraint is that
+every share must be processed exactly once by all aggregators. Data integrity
+constraints may be at odds with the threat model if meeting the constraints
+requires replaying data.
+
+Aggregator operators should expect to encounter invalid inputs during regular
+operation due to misconfigured or malicious clients. Low volumes of errors are
+tolerable; the input-verification protocol and AFEs are robust in the face of
+malformed data. Aggregators may need to detect and mitigate statistically
+significant floods of invalid or identical inputs that affect accuracy, e.g.,
+denial of service (DoS) events.
+
+Certain classes of errors do not exist in the input-validation protocol
+considered in this document. For example, packet loss errors when clients make
+requests directly to aggregators are not relevant when the leader proxies
+requests and controls the schedule for signaling aggregation rounds.
```

Comment:
I realize this is copy-pasted, but I'm questioning the value of this paragraph at this point. Is it saying anything that's new and not obvious?

Reply:
Nope, we can delete.